parsing/dialects/intermediate/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ValidatorTest.java

```diff
@@ -101,4 +101,32 @@
         + "FROM `GHI` AS `GHI`";
     sql(sql).rewritesTo(expected);
   }
+
+  @Test public void testCreateTableNoColumns() {
+    String ddl = "create table foo";
+    String query = "select * from foo";
+    sql(ddl).ok();
+    sql(query).type("RecordType() NOT NULL");
+  }
+
+  @Test public void testCreateTableWithColumns() {
```

Comment:
Suggest breaking all of these into their own individual test cases. Even though that results in duplicating the same setup code, it simplifies the testing logic and makes it easier for a future developer to reason about what caused a test to break.

Reply:
Sure, SGTM.